### ISSUE TITLE
GUI: support for RTL

### ISSUE THREAD

**Issue by heatherleaf** [MUSTE-Project/MULLE#47]
Perhaps right-to-left works out of the box, but we should test. For this we first need a toy grammar in an RTL language

**Comment by fredefox** [comment]
Should be achievable with a combination of https://developer.mozilla.org/en-US/docs/Web/CSS/direction and extending the data model to include this bit of information.

**Comment by heatherleaf** [comment]
I've updated Exemplum with Arabic, which is RTL. Seems to work mostly fine, except two things.

### Selections at the very start and end

```
$ muste cli -G exemplum/Exemplum -L ExemplumAra --print-nodes " يَقْرَأُ مَلِكٌ جوَيِّدٌ كِتابً "
```

As you can see here, the `{3-3}` insertion is fine, but `{0-0}` should be to the right and `{4-4}` should be to the left (but it's the other way around):
```
{0-0}: [ ] يَقْرَأُ مَلِكٌ جوَيِّدٌ كِتابً                   [ ] read_V king_N good_A book_N
[ الآن ] يَقْرَأُ مَلِكٌ جوَيِّدٌ كِتابً                     [ now_Adv+today_Adv ] read_V king_N good_A book_N
(...)

{3-3}: يَقْرَأُ مَلِكٌ جوَيِّدٌ [ ] كِتابً                   read_V king_N good_A [ ] book_N
يَقْرَأُ مَلِكٌ جوَيِّدٌ [ الآن ] كِتابً                     read_V king_N good_A [ now_Adv+today_Adv ] book_N
(...)

{4-4}: يَقْرَأُ مَلِكٌ جوَيِّدٌ كِتابً [ ]                   read_V king_N good_A book_N [ ]
يَقْرَأُ مَلِكٌ جوَيِّدٌ كِتابً [ أَزرَقَ ]                  read_V king_N good_A book_N [ blue_A ]
(...)
```

The same for replacements: `{1-2}` and `{2-3}` are fine, but not `{0-1}` or `{3-4}`:
```
{0-1}: [ يَقْرَأُ ] مَلِكٌ جوَيِّدٌ كِتابً                   [ read_V ] king_N good_A book_N
[ يَحْبُِّ ] مَلِكٌ جوَيِّدٌ كِتابً                          [ love_V ] king_N good_A book_N
(...)

{1-2}: يَقْرَأُ [ مَلِكٌ ] جوَيِّدٌ كِتابً                   read_V [ king_N ] good_A book_N
تَقْرَأُ [ ؟مٌ ] جوَيِّدَةٌ كِتابً                           read_V [ mother_N ] good_A book_N
(...)

{2-3}: يَقْرَأُ مَلِكٌ [ جوَيِّدٌ ] كِتابً                   read_V king_N [ good_A ] book_N
يَقْرَأُ المُلُوكُ [ ] كِتابً                                read_V king_N [ ] book_N
(...)

{3-4}: يَقْرَأُ مَلِكٌ جوَيِّدٌ [ كِتابً ]                   read_V king_N good_A [ book_N ]
يَقْرَأُ مَلِكٌ جوَيِّدٌ [ ؟مً ]                             read_V king_N good_A [ mother_N ]
(...)
```

**Comment by heatherleaf** [comment]
Similar for the GUI and insertions:
![image](https://user-images.githubusercontent.com/1324196/46306905-d2220f00-c5b5-11e8-817e-31cb49c8691d.png)
![image](https://user-images.githubusercontent.com/1324196/46306919-dbab7700-c5b5-11e8-87ee-de328a87f494.png)

But there is no problem with replacements:
![image](https://user-images.githubusercontent.com/1324196/46307007-11e8f680-c5b6-11e8-99b5-99266af6994d.png)
![image](https://user-images.githubusercontent.com/1324196/46307035-1f9e7c00-c5b6-11e8-9731-c8e2099f6149.png)

(Safari, Firefox and Chrome all look the same)

**Comment by heatherleaf** [comment]
Probably we need to add Unicode RTL and LTR marks at the beginning and end. If we're lucky it's enough to do that on the server side and it will automagically appear in the browser too... (but probably it's not that easy)

https://www.w3.org/TR/WCAG20-TECHS/H34.html

**Comment by fredefox** [comment]
At the Safari bug: I don't have access to a MAC unfortunately...

**Comment by heatherleaf** [comment]
Simple solution: add css `direction:rtl` to the `<div class="sentence">` node corresponding to the sentence. Also: add a setting "direction" to `lessons.yaml`

**Comment by fredefox** [comment]
Closed in 63d31a6

**Comment by heatherleaf** [comment]
need to set `direction:rtl` on each menu item too

**Comment by fredefox** [comment]
Closed in 3531226
